queue/mock_test.go

```diff
@@ -0,0 +1,289 @@
+package queue
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/utility"
+	"github.com/google/uuid"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func init() {
+	mockJobCounters = &mockJobRunEnv{}
+	registry.AddJobType("mock", func() amboy.Job { return newMockJob() })
+	registry.AddJobType("sleep", func() amboy.Job { return newSleepJob() })
+	registry.AddJobType("retryable", func() amboy.Job { return &mockRetryableJob{} })
+}
+
+// mockRetryableQueue provides a mock implementation of a remoteQueue whose
+// runtime behavior is configurable. Methods can be mocked out; otherwise, the
+// default behavior is identical to that of its underlying remoteQueue.
+type mockRemoteQueue struct {
+	remoteQueue
+
+	// Mockable methods
+	putJob        func(remoteQueue, context.Context, amboy.Job) error
```

Comment:
Some of these funcs have named arguments and others don't, I think it should be consistent. Also, small nit but can you put `ctx context.Context` as the first arg?

Reply:
I only named the ones that had non-obvious arguments (i.e. what does a string mean when passed to Get), but I can name all of them for consistency.

Moved `context.Context`.

Reply:
You don't have to name everything, that is fine. But `context.Context` isn't moved, maybe you forgot to push your changes?

Reply:
You're probably viewing the outdated one. The new commit is [this](https://github.com/mongodb/amboy/pull/184/commits/656c7747beadb6887d20bd3e934e5283eec11ea3#diff-3a3bd2a2c3807e9ea4810a5363bc023f9b5441c1f6de21eaf11a3fc0c8a4e3dfR34).